tutorons/css/explain.py

```diff
@@ -24,15 +29,43 @@
 def explain(selector):
     return explainer.explain(selector)
 
+def filter_non_ascii(s):
+    if ord(s) > 127:
+        return ' '
+    return s
 
-class CssSelectorExtractor(object):
+class JavascriptSelectorExtractor(object):
 
     def __init__(self):
         self.js_string_extractor = JavascriptStringExtractor()
 
     def extract(self, node):
+
         regions = self.js_string_extractor.extract(node)
         valid_regions = [r for r in regions if is_selector(r.string)]
+        
+        return valid_regions
+
+class StylesheetSelectorExtractor(object):
+
+    def extract(self, node):
+        textfield = ''.join(map(filter_non_ascii, node.text))
+        parser = tinycss.make_parser()
+        stylesheet = parser.parse_stylesheet(textfield)
+        print(stylesheet)
+
+        if len(stylesheet.rules) == 0:
```

Comment:
Unnecessary if-statement here.  You initialize an empty list of regions, and then add a region for every rule found.  So why not just always execute the body of the "else" ?